### ISSUE TITLE
It is possible to overwrite methods in R6 objects

### ISSUE THREAD

**Issue by richfitz** [r-lib/R6#19]
Once an R6 object has been constructed, it is possible to overwrite its methods, which seems undesirable:

``` r
Thing <- R6::R6Class("Thing",
                     public=list(
                       a=1L,
                       f=function() a))
obj <- Thing$new()
obj$f() # 1L
obj$f <- NULL # is allowed
obj$f() # Error: attempt to apply non-function
```

And of course, putting the method "back" doesn't work because the environment of that function is wrong:

``` r
obj$f <- function() a
obj$f() # Error: object 'a' not found
```

Is there any way of easily preventing this happen without sacrificing the speed goals of R6?

**Comment by mmuurr** [comment]
I realized this was closed 6(!) years ago ... but I'm wondering how one would "unlock the bindings themselves". I suddenly find myself in a position where I'd like to update a method _after_ an object has been created (i.e. an update on the instance). I tried creating the generator with `lock_objects = FALSE`, but I'm still getting the `cannot change value of locked binding for 'foo'` error. Is this doable with current R6 implementation?
